src/manifest.ts

```diff
@@ -1260,76 +1261,67 @@ export class Manifest {
-          continue;
+          this.logger.debug('Branches in sync, safe to re-align');
+          safeToRealign = true;
         }
       } catch (err: unknown) {
+        // if a branch is not found, it is likely that the release branch has been deleted. In this case just ignore and
+        // continue with the next check
         if (isOctokitRequestError(err) && err.status === 404) {
-          // note: this really should trigger an alert of some sort, but that's not yet a thing
-          this.logger.error(
-            `Branch comparison between '${pr.headBranchName}' and '${branchName.changesBranch}' failed due to a missing branch. As a result branches '${pr.baseBranchName}' and '${branchName.changesBranch}' won't be re-aligned which may result in git conflicts when the next release PR is created. Note: the release branch '${pr.headBranchName}' used for the PR ${pr.number} has likely been deleted manually before the release process could run, resulting in this error.`
+          this.logger.debug(
+            `Could not compare branches '${pr.headBranchName}' and '${branchName.changesBranch}' due to a missing branch. Continue with the next check`
           );
-          continue;
+        } else {
+          throw err;
         }
       }
 
-      // then check if changes-branch has already been synced with the base branch (that's the case if the release
-      // process runs twice in a row for example)
-      if (
-        await this.github.isBranchASyncedWithB(
-          branchName.changesBranch,
-          pr.baseBranchName
-        )
-      ) {
-        continue;
-      }
-
-      throw new Error(
-        'Race condition detected. The ref branch used to create the release pull request branch received new commits. Cannot safely reset branch.'
-      );
-    }
-  }
-
-  private async alignPullRequestsChangesBranch(pullRequests: PullRequest[]) {
-    for (const pr of pullRequests) {
-      const branchName = BranchName.parse(pr.headBranchName);
-      // we only care about pull requests with an associated changes-branch
-      if (!branchName?.changesBranch) {
-        continue;
-      }
-      this.logger.info(
-        `Aligning pull request branches for PR #${pr.number}, changes branch ${branchName.changesBranch}`
-      );
+      // then check if changes-branch has already been synced with the base branch, in which case we don't need to do
+      // anything
       if (
-        await this.github.isBranchASyncedWithB(
+        !safeToRealign &&
+        (await this.github.isBranchASyncedWithB(
           branchName.changesBranch,
           this.targetBranch
-        )
+        ))
       ) {
         this.logger.debug(
-          `Branches ${branchName.changesBranch} and ${this.targetBranch} already in sync, do nothing`
+          `Checking if ${branchName.changesBranch} is synced with ${this.targetBranch}...`
         );
+        this.logger.debug('Branches already in sync, no need to re-align');
         continue;
```

Comment:
Should we also handle the case where `next` has been deleted? 🤔

Reply:
Let's not worry about that for now